docs/concepts/json.md

```diff
@@ -60,6 +60,156 @@ with one noticeable enhancement being that `jiter` supports deserialization of `
 in the original JSON input which contained the invalid value.
 
+### Partial JSON Parsing
+
+**Starting in v2.7.0**, Pydantic's [JSON parser](https://docs.rs/jiter/latest/jiter/) offers support for partial JSON parsing, which is exposed via [`pydantic_core.from_json`][pydantic_core.from_json]. Here's an example of this feature in action:
+
+```py
+from pydantic_core import from_json
+
+partial_json_data = '["aa", "bb", "c'  # (1)!
+
+try:
+    result = from_json(partial_json_data, allow_partial=False)
+except ValueError as e:
+    print(e)  # (2)!
+    #> EOF while parsing a string at line 1 column 15
+
+result = from_json(partial_json_data, allow_partial=True)
+print(result)  # (3)!
+#> ['aa', 'bb']
+```
+
+1. The JSON list is incomplete - it's missing a closing `"]`
+2. When `allow_partial` is set to `False` (the default), a parsing error occurs.
+3. When `allow_partial` is set to `True`, part of the input is deserialized successfully.
+
+This also works for deserializing partial dictionaries. For example:
+
+```py
+from pydantic_core import from_json
+
+partial_dog_json = '{"breed": "lab", "name": "fluffy", "friends": ["buddy", "spot", "rufus"], "age}'
+dog_dict = from_json(partial_dog_json, allow_partial=True)
+print(dog_dict)
+#> {'breed': 'lab', 'name': 'fluffy', 'friends': ['buddy', 'spot', 'rufus']}
+```
+
+!!! tip "Validating LLM Output"
+    This feature is particularly beneficial for validating LLM outputs.
+    We've written some blog posts about this topic, which you can find [here](https://blog.pydantic.dev/blog/category/llms/).
+
+In future versions of Pydantic, we expect to expand support for this feature through either Pydantic's other JSON validation functions
+([`pydantic.main.BaseModel.model_validate_json`][pydantic.main.BaseModel.model_validate_json] and
+[`pydantic.type_adapter.TypeAdapter.validate_json`][pydantic.type_adapter.TypeAdapter.validate_json]) or model configuration. Stay tuned 🚀!
+
+For now, you can use [`pydantic_core.from_json`][pydantic_core.from_json] in combination with [`pydantic.main.BaseModel.model_validate`][pydantic.main.BaseModel.model_validate] to achieve the same result. Here's an example:
+
+```py
+from pydantic_core import from_json
+
+from pydantic import BaseModel
+
+
+class Dog(BaseModel):
+    breed: str
+    name: str
+    friends: list
+
+
+partial_dog_json = '{"breed": "lab", "name": "fluffy", "friends": ["buddy", "spot", "rufus"], "age}'
+dog = Dog.model_validate(from_json(partial_dog_json, allow_partial=True))
+print(repr(dog))
+#> Dog(breed='lab', name='fluffy', friends=['buddy', 'spot', 'rufus'])
+```
+
+!!! example
+    We recommend the following settings to ensure reliable parsing of partial JSON data:
+
+    1. For this to work reliably, fields on the model should have default values or be optional.
+    2. Set `on_error='omit'` in the core schema for types that can fail. This is the default behavior for Pydantic fields.
```

Comment:
Although this would work, it would mean any error on that field would cause the default value to be used, so I'm not sure it's a good idea to suggest this.